src/test/java/co/rsk/federate/signing/hsm/message/AdvanceBlockchainMessageTest.java

```diff
@@ -15,16 +16,37 @@
 public class AdvanceBlockchainMessageTest {
 
     @Test
-    public void getBlockHeaders_ok() {
+    public void getBlockHeaders_BrothersList_ok() {
         byte[] encodedBlockHeader1 = new byte[]{ 1 };
         BlockHeader blockHeader1 = mock(BlockHeader.class);
         when(blockHeader1.getFullEncoded()).thenReturn(encodedBlockHeader1);
 
-        List<BlockHeader> blockHeaders = Collections.singletonList(blockHeader1);
-        AdvanceBlockchainMessage message = new AdvanceBlockchainMessage(blockHeaders);
+        byte[] encodedBlockHeader2 = new byte[]{ 2 };
+        BlockHeader blockHeader2 = mock(BlockHeader.class);
+        when(blockHeader2.getFullEncoded()).thenReturn(encodedBlockHeader2);
+
+        byte[] encodedBlockHeader3 = new byte[]{ 3 };
+        BlockHeader blockHeader3 = mock(BlockHeader.class);
+        when(blockHeader3.getFullEncoded()).thenReturn(encodedBlockHeader3);
+
+        byte[] encodedBlockHeader4 = new byte[]{ 4 };
+        BlockHeader blockHeader4 = mock(BlockHeader.class);
+        when(blockHeader4.getFullEncoded()).thenReturn(encodedBlockHeader4);
+
+        List<BlockHeader> blockHeaders = Arrays.asList(blockHeader1, blockHeader2, blockHeader3);
+        List<List<BlockHeader>> brothersList = Arrays.asList(Arrays.asList(blockHeader1, blockHeader2), Collections.emptyList(), Collections.singletonList(blockHeader4));
+        AdvanceBlockchainMessage message = new AdvanceBlockchainMessage(blockHeaders, brothersList);
+
+        Assert.assertEquals(3, message.getBlockHeaders().size());
+        Assert.assertEquals(3, message.getBrothersList().size());
+        // Inverted order
+        Assert.assertEquals(Hex.toHexString(encodedBlockHeader3), message.getBlockHeaders().get(0));
+        Assert.assertEquals(Hex.toHexString(encodedBlockHeader2), message.getBlockHeaders().get(1));
+        Assert.assertEquals(Hex.toHexString(encodedBlockHeader1), message.getBlockHeaders().get(2));
 
-        Assert.assertEquals(1, message.getBlockHeaders().size());
-        Assert.assertEquals(Hex.toHexString(encodedBlockHeader1), message.getBlockHeaders().get(0));
+        Assert.assertEquals(2, message.getBrothersList().get(0).length);
+        Assert.assertEquals(0, message.getBrothersList().get(1).length);
+        Assert.assertEquals(1, message.getBrothersList().get(2).length);
```

Comment:
Shouldn't we also check we have the expected brothers for each header?

Reply:
Done